packages/calcite-components/src/components/input-time-picker/input-time-picker.tsx

```diff
@@ -586,9 +579,9 @@ export class InputTimePicker
     syncHiddenFormInput("time", this, input);
   }
 
-  private delocalizeTimeString(value: string): string {
+  private delocalizeTimeString(value: string, locale?: SupportedLocale): string {
     // we need to set the corresponding locale before parsing, otherwise it defaults to English (possible dayjs bug)
-    dayjs.locale(this.messages._lang.toLowerCase());
+    dayjs.locale(locale || this.getSupportedDayjsLocale(this.messages._lang.toLowerCase()));
```

Comment:
It doesn't seem like `locale` is passed in any usage of `delocalizeTimeString`. If this is the case, let's remove it.